app/controllers/api/v1/venues_controller.rb

```diff
@@ -0,0 +1,59 @@
+# :reek:InstanceVariableAssumption :reek:NilCheck
+class Api::V1::VenuesController < ApplicationController
+  before_action :organizer?, only: %i[create update destroy]
+  before_action :set_venue, only: %i[show update destroy]
+
+  def index
+    @venues = Venue.all.reverse
+    render json: @venues
+  end
+
+  def show
+    if @venue
+      render json: @venue
+    else
+      render json: { status: :not_found }
+    end
+  end
+
+  def create
+    @venue = Venue.new(venue_params)
+    if @venue.save
+      render json: @venue, status: :created
+    else
+      render json: @venue.errors.full_messages, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @venue.update_attributes(venue_params)
+      render json: @venue
+    else
+      render json: @venue.errors.full_messages, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    if @venue&.destroy
+      render json: { status: :deleted }
```

Comment:
if you return only status, you will set it in header. Instead this line you can do next
```
head :deleted
```

Reply:
done